viam-cartographer/src/slam_service/slam_service_test.cc

```diff
@@ -360,7 +334,7 @@ BOOST_AUTO_TEST_CASE(GetNextDataFileOffline) {
         utils::createTmpDirectoryAndAddFiles(data_files, map_files);
     slamService.path_to_data = tmp_dir.string() + "/data";
 
-    for (int i = 0; i < data_files.size(); i++) {
+    for (int i = 0; i < data_files.size() - 1; i++) {
```

Comment:
We now ignore the last file in offline mode because it's possible that it's still being written to.